docs/application/dotnet/guides/nui/texteditor.md

```diff
@@ -0,0 +1,113 @@
+# TextEditor
+
+The `TextEditor` class provides a control that allows multi-line text editing. It is similar to the [TextField](./textfield.md) control, where a different formatting can be applied to different parts of the text.  You can change, for example, the font color, font style, point size, and font family.
+
+The `TextEditor` also supports markup, and text can be scrolled vertically within it.
+
+**Figure: TextEditor**
+
+![TextEditor](./media/dali_texteditor.png)
+
+In this tutorial, the following subjects are covered:
+
+[TextEditor events](#1)<br>
+[Creating a TextEditor](#2)<br>
+[TextEditor Properties](#3)<br>
+
+<a name="1"></a>
+## TextEditor events
+
+The following table lists the basic signals provided by the `TextEditor` class.
+
+**Table: TextEditor input signals**
+
+| Input signal         | Description                              |
+| -------------------- | ---------------------------------------- |
+| `TextChanged`        | Emitted when the text changes.           |
+| `ScrollStateChanged` | Emitted when TextEditor scrolling is started or finished. |
+
+<a name="2"></a>
+## Creating a TextEditor
+
+The following basic example shows how to create a `TextEditor` object:
+
+```
+// Create a TextEditor instance
+Window window = Window.Instance;
+TextEditor editor = new TextEditor();
+editor.Position2D = new Position2D(10, 700);
+editor.Size2D = new Size2D(400, 90);
+editor.BackgroundColor = Color.Red;
+editor.PointSize = 20;
+editor.TextColor = Color.White;
+editor.Text = "This is a multiline text.\n I can write several lines.\n"
+window.Add(editor);
+```
+
+<a name="3"></a>
+## TextEditor Properties
+
+You can modify the `TextEditor` appearance and behavior through its properties.
+
+The following table lists the available `TextEditor` properties.
+
+**Table: TextEditor properties**
+
+| Property                           | Type        | Description                              |
+| ---------------------------------- | ----------- | ---------------------------------------- |
```

Comment:
Please add period at the end of the sentence for all the Description under table properties.

Reply:
Done.